lab2/kursinis.tex

```diff
@@ -47,10 +47,29 @@ \section{Context}
+    \item \textbf{Integration with GPT API}: The project integrates with Chat GPT, providing users with additional functionalities. This integration allows the system to store user chat history and respond to user inquiries using the capabilities of Chat GPT. Users can leverage this feature to seek assistance, clarify doubts, or engage in interactive conversations within the system.
+    \end{itemize}
+
+\subsection{Tech stack and services}
+
+The EduPal system was created using a modern technology stack to support its functionalities. The basic structure of the system:
+
+\begin{itemize}
+    \item \textbf{Frontend}: The frontend of the system is developed using the React framework and JavaScript with custom CSS. Frontend allows communication with the server via the fetch function, which utilizes the Fetch API to send HTTP requests to the particular API endpoint in the backend.
+    \item \textbf{Backend}: The system is built using ASP.NET Core and uses Entity Framework Core which helps following monolithic architecture. It incorporates the Repository Pattern for handling data access, particularly for ASP.NET Identity user management. The backend is built using .NET 7.0 and follows a RESTful API architecture, enabling it to handle HTTP requests and responses. Additionally, WebSocket communication is facilitated by SignalR and is used for ChatHub endpoint which, communication via comment section. Passwords are stored directly as plaintext within the database. Logs are managed by the interceptor, tracking database changes. Also, tokens are generated upon successful user authentication. These tokens contain user information and are used for subsequent authentication and authorization requests between the client and the server.
+    \item\textbf{Data Storage}: Conspectuses are stored locally within the application's file system. Upon upload, each conspectus is saved as a separate file in a directory named "Files" relative to the application's working directory. When users request to download a conspectus, the application retrieves the corresponding file from this directory and streams it back as downloadable content. Also PostgreSQL is used to store information about the users. The use of Entity Framework Core provides tighter integration with the backend for database operations.
+    \item \textbf{OpenAI Service}: The system uses OpenAI API, to provide Chat GPT services. API helps connecting to the server and sending any user inqueries. The prompt sent to Open AI is modified so that length of the response would be reasonable.
+    \item \textbf{Tests}: System is tested using a combination of unit and integration tests.
+    \item \textbf{Version Control}: Git is used for code publishing and reviewing processes.
```

Comment:
Git does not have reviewing processes. Name section Version Control and Hosting and specify that git is used and gitlab for hosting and project collaboration (issues, prs)